### ISSUE TITLE
How to extract just bounding boxes? (Only interested in RPN)

### ISSUE THREAD

**Issue by edualvarado** [matterport/Mask_RCNN#1661]
Hello,

I am trying to display on the final output just the bounding box coming out from the RPN, in order to only locate the object and speed up the algorithm. I am not interested in classification so far.

Is the code prepared to do so in a fast and efficient way? I am searching but I do not get how I could do it, even commenting the lines which perform classification. Can you help me with this? 

Thank you!

**Comment by dimartinot** [comment]
Unfortunately, I doubt this could be possible without restructuring everything... If you're willing to do so, you could still copy the code for the RPN (including the loss functions and etc...) then insert it into your own project. However, you would have to train it yourself and to have, therefore, ground-truth data for the bounding boxes.

I do not know if it will help you but you can already retrieve the RPN Region Proposals as a part of the output of the model (_model.py, l.2054)_:
````python

 model = KM.Model([input_image, input_image_meta, input_anchors],
                             [detections, mrcnn_class, mrcnn_bbox, mrcnn_mask, rpn_rois, rpn_class, rpn_bbox],
````

These boxes are stored in the rpn_bbox variable retrievable by the model.predict(...) method.

Also, I have myself developed an RPN. I definitly do not think that it will be as good as the one from Matterport but you could try it out if you want ! It is stored there : [link](https://github.com/dimartinot/bunch_of_networks/blob/master/Keras_Impl/Networks/RPN/rpnModel.py)

**Comment by karthik9319** [comment]
r['rois'] will have your bounding box co-ordinates
```
def detect_and_color_splash:
     r = model.detect([image], verbose=1)[0]
     print (r['rois'][0][0], r['rois'][0][1], r['rois'][0][2], r['rois'][0][3])
```
